### PR TITLE
Github issue 8304: remove obsolete sample payment module from payment pages

### LINKED ISSUE #8304: Update references to Payment sample module
# General issue

You are providing old version support of Magento and archived repo for code base on creating a custom payment method

## Description:

Affected page: /guides/v2.4/payments-integrations/base-integration/integration-intro.html

In the following text on the documentation:

To simplify the development of new payment integration, Magento developed the [Payment sample module](https://github.com/magento/magento2-samples/tree/master/sample-module-payment-gateway). It contains all required infrastructure and you can use it as a starting point.

The link provided on **Payment sample module** link points to an outdated repo (even archived and 5yo) when specifying that the documentation if for Magento 2.4

## Possible solutions:

Create or Point such link to a recent version of the Payment sample module

## Additional information:

NOPE

### PR DESCRIPTION
**Pull request by Viper9x**
Fixes #8304
Remove obsolete sample payment module from payment pages

## Purpose of this pull request

Remove obsolete sample payment module from payment pages

## Affected DevDocs pages

- https://devdocs.magento.com/guides/v2.4/payments-integrations/base-integration/integration-intro.html
- https://devdocs.magento.com/guides/v2.4/payments-integrations/base-integration/module-configuration.html
- https://devdocs.magento.com/guides/v2.3/payments-integrations/base-integration/integration-intro.html
- https://devdocs.magento.com/guides/v2.3/payments-integrations/base-integration/module-configuration.html

### CODE REVIEW

**Review by keharper — CHANGES_REQUESTED** [review]
@Viper9x I'm putting a block on this PR before someone else merges it.

The links to the dead repo were left in the docs on purpose. They provide a skeleton of what you need to develop a payment module.

I need to ask around to determine the next step. Most likely, I'll ask you to back out these changes and add a note stating the linked repo contains unsupported code.

Semi-related PR: https://github.com/magento/devdocs/pull/7417

**Comment by Viper9x** [comment]
thanks @keharper
Total agree, I think it would better if we can keep the link to sample module for reference, and add a note to say the module is now not supported.

File: src/guides/v2.3/payments-integrations/base-integration/integration-intro.md [hunk 1]
```diff
@@ -19,9 +19,6 @@ The Magento payment provider gateway allows creating secure and PCI-compliant in
 The topics in this chapter explain how to add an integration with a custom payment service provider (in other words, add a new payment method) and implement the authorize payment action for this [payment method](https://glossary.magento.com/payment-method). For illustration, we use code
 samples from the [Braintree]({{ site.mage2bloburl }}/{{ page.guide_version }}/app/code/Magento/Braintree) payment integration.
 
-To simplify the development of a new payment integration, Magento developed the [Payment sample module](https://github.com/magento/magento2-samples/tree/master/sample-module-payment-gateway).
```
Comment (keharper) [inline src/guides/v2.3/payments-integrations/base-integration/integration-intro.md:22]: Revert this deletion and change it to something like

{:.bs-callout-info}
You can also view the [Payment sample module](https://github.com/magento/magento2-samples/tree/master/sample-module-payment-gateway) in the `magento/magento2-samples` 
 repository to understand the underlying principles.  However, be aware that this code is NOT supported.

File: src/guides/v2.3/payments-integrations/base-integration/module-configuration.md [hunk 2]
```diff
@@ -10,8 +10,6 @@ functional_areas:
 
 For the sake of compatibility, upgradability and easy maintenance, do not edit the default Magento code; add your customizations in a separate [module](https://glossary.magento.com/module).
 
-You can use the [sample Magento_SamplePaymentGateway module](https://github.com/magento/magento2-samples/tree/master/sample-module-payment-gateway) files as basis for your custom module structure and files.
```
Comment (keharper) [inline src/guides/v2.3/payments-integrations/base-integration/module-configuration.md:13]: Revert this deletion and replace with something like:

{:.bs-callout-info}
You can use the [sample Magento_SamplePaymentGateway module](https://github.com/magento/magento2-samples/tree/master/sample-module-payment-gateway) files as the basis for your custom module structure and files. However, be aware that this code is NOT supported.

**Review by keharper — CHANGES_REQUESTED** [review]
@Viper9x We'll keep the links to the old repo.

**Review by keharper — APPROVED** [review]

**Comment by keharper** [comment]
@magento import pr to magento-devdocs/devdocs

**Comment by m2-github-services** [comment]
@keharper the Pull Request is successfully imported.

**Pull request merged**